lib/dry/types/schema.rb

```diff
@@ -284,6 +284,19 @@ def lax
         Lax.new(schema(keys.map(&:lax)))
       end
 
+      # Merges given schema keys into current schema.
+      #
+      # A new instance is returned.
+      #
+      # @param schema [Schema]
+      # @return [Schema]
+      #
+      # @api public
+      def merge(schema)
+        with(keys: keys + schema.keys)
+      end
+      alias_method :compose, :merge
```

Comment:
Let's not add it for now. Conceptually, it's okay, but from my experience, the fewer methods dry-types has, the better. That said, we can add it later if we see it'll be useful in other gems.

Reply:
That's ok. I just found sweet having `dry-struct`'s `#compose` (the other PR I referenced) calling underneath schema's `#compose`, but it's not such a big deal :smile: